app/controllers/vote_posts_controller.rb

```diff
@@ -16,10 +16,12 @@ def create
     @vote_post.user = current_user
 
     if VoteService.user_vote(@vote_post)
-      redirect_to votes_path, notice: alert_create(VotePost)
+      flash[:notice] = alert_create(VotePost)
     else
-      render :new, status: 422
+      flash[:alert] = I18n.t('vote_post.already_voted')
```

Comment:
Is this always the case when VoteService returns false? What happens with other errors?